src/V3CfgBuilder.cpp

```diff
@@ -0,0 +1,260 @@
+    void visit(AstWhile* nodep) override {
+        if (!m_cfgp) return;
+
+        // Create header/body/continuation blocks
+        BasicBlock& headBB = addBasicBlock();
+        BasicBlock& bodyBB = addBasicBlock();
+        BasicBlock& contBB = addBasicBlock();
+        addEdge(*m_currBBp, headBB, EdgeKind::Jump);
+        addEdge(headBB, bodyBB, EdgeKind::ConditionTrue);
+        addEdge(headBB, contBB, EdgeKind::ConditionFalse);
+
+        // The While goes in the header block - semantically the condition check only ...
+        headBB.stmtps().emplace_back(nodep);
+
+        // Build the body
+        m_currBBp = &bodyBB;
+        iterateAndNextConstNull(nodep->stmtsp());
+        iterateAndNextConstNull(nodep->incsp());
+        if (!m_cfgp) return;
+        if (m_currBBp) addEdge(*m_currBBp, headBB, EdgeKind::Jump);
+
+        // Set continuation
+        m_currBBp = &contBB;
+    }
+    void visit(AstJumpBlock* nodep) override {
+        if (!m_cfgp) return;
+
+        // Don't acutally need to add this 'nodep' to any block - but we could later if needed
+
+        // Create continuation block
+        BasicBlock& contBB = addBasicBlock();
+        const bool newEntry = m_jumpBlockContp.emplace(nodep, &contBB).second;
+        UASSERT_OBJ(newEntry, nodep, "AstJumpBlock visited twice");
+
+        // Build the body
+        iterateAndNextConstNull(nodep->stmtsp());
+        if (!m_cfgp) return;
+        if (m_currBBp) addEdge(*m_currBBp, contBB, EdgeKind::Jump);
+
+        // Set continuation
+        m_currBBp = &contBB;
+    }
+    void visit(AstJumpGo* nodep) override {
+        if (!m_cfgp) return;
+
+        // Non-representable if not last in statement list (V3Const will fix this later)
+        if (nodep->nextp()) {
+            m_cfgp.reset();
+            return;
+        }
+
+        // Don't acutally need to add this 'nodep' to any block - but we could later if needed
+
+        // Make current block go to the continuation of the JumpBlock
+        addEdge(*m_currBBp, *m_jumpBlockContp.at(nodep->blockp()), EdgeKind::Jump);
+
+        // There should be no statements after a JumpGo!
+        m_currBBp = nullptr;
+    }
+
+    // Representable non control-flow statements
+    void visit(AstAssign* nodep) override { simpleStatement(nodep, !nodep->timingControlp()); }
+    void visit(AstComment*) override {}  // ignore entirely
+    void visit(AstDisplay* nodep) override { simpleStatement(nodep); }
+    void visit(AstFinish* nodep) override { simpleStatement(nodep); }
+    void visit(AstStmtExpr* nodep) override { simpleStatement(nodep); }
+    void visit(AstStop* nodep) override { simpleStatement(nodep); }
+
+    // Non-representable nodes
+    void visit(AstAssignDly* nodep) override { nonRepresentable(); }
+    void visit(AstCase* nodep) override { nonRepresentable(); }  // V3Case will eliminate
+    void visit(AstCReset* nodep) override { nonRepresentable(); }
+    void visit(AstDelay* nodep) override { nonRepresentable(); }
```

Comment:
Can you use an existing virtual bool method to determine this?  Should there be one?  Either way well document what makes these special.

Reply:
> Can you use an existing virtual bool method to determine this? Should there be one?

I kind of read that as "would you like to distribute these related lines clearly collected next to each other over 5000 lines of unrelated code?" ;) None of those virtual functions should exist, but that's another matter.

I will add comments, but the reason these are special is because these statements have some special semantics that makes data-flow analysis unclear when they are involved (at least haven't thought about it). Point again here is to be conservative and give up on everything that is not known to be handled correctly. We can always support more stuff if the client code requires it, but don't want to chase bugs because we assume something is correct without ever actually testing it.